packages/babel-parser/src/parser/util.js

```diff
@@ -259,6 +259,24 @@ export default class UtilParser extends Tokenizer {
       this.raise(doubleProto, Errors.DuplicateProto);
     }
   }
+
+  /**
+   * Test if current token is a literal property name
+   * https://tc39.es/ecma262/#prod-LiteralPropertyName
+   * LiteralPropertyName:
+   *   IdentifierName
+   *   StringLiteral
+   *   NumericLiteral
+   *   BigIntLiteral
+   */
+  isLiteralPropertyName(): boolean {
+    return (
+      this.match(tt.name) ||
+      this.match(tt.string) ||
+      this.match(tt.num) ||
+      this.match(tt.bigint)
```

Comment:
Q: things like `true` or `null` fall in the `tt.name` case, right?

Reply:
The fall in the special `tt._true` and `tt._false`. I move all keywords to `isLiteralPropertyName` since they are valid identifier name.

Reply:
Nice catch!